kong/runloop/handler.lua

```diff
@@ -1419,6 +1425,16 @@ return {
       if var.http_proxy_connection then
         clear_header("Proxy-Connection")
       end
+
+      -- X-Kong-Request-Id upstream header
+      local rid, rid_get_err = request_id_get()
+      if not rid then
+        log(WARN, "failed to get Request ID: ", rid_get_err)
```

Comment:
just a reminder: If an error occurs, the log will be flooded as a large number of requests arrive.  If this error log is designed to capture some potential software bug, you could resolve this comment directly.

Reply:
yes this is only meant to capture and log a scenario that is not really supposed to occur, i.e. the request ID not being available during a valid phase. I would never expect this to be logged but prefer to keep it just in case.